tools/run_tests/run_xds_tests.py

```diff
@@ -1762,81 +1780,88 @@ def _delay(pct):
+        deadline = time.time() + _WAIT_FOR_URL_MAP_PATCH_SEC
+        logger.info(
+            'Attempting for %d seconds until received the expected distribution',
+            _WAIT_FOR_URL_MAP_PATCH_SEC)
+        attempt_counter = 0
+        while True:
+            attempt_counter += 1
+            try:
+                for (testcase_name, client_config,
+                     expected_results) in test_cases:
+                    logger.info('starting case %s: attempt %d', testcase_name,
+                                attempt_counter)
+
+                    client_config['metadata'] = [
+                        (messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+                         testcase_header, testcase_name)
+                    ]
+                    client_config['rpc_types'] = [
+                        messages_pb2.ClientConfigureRequest.RpcType.UNARY_CALL,
+                    ]
+                    configure_client(**client_config)
+                    # wait a second to help ensure the client stops sending RPCs with
+                    # the old config.  We will make multiple attempts if it is failing,
+                    # but this improves confidence that the test is valid if the
+                    # previous client_config would lead to the same results.
+                    time.sleep(1)
+                    # Each attempt takes 10 seconds; 20 attempts is equivalent to 200
+                    # second timeout.
+                    before_stats = get_client_accumulated_stats()
+                    if not before_stats.stats_per_method:
+                        raise ValueError(
+                            'stats.stats_per_method is None, the interop client stats service does not support this test case'
+                        )
+                    logger.info('%s: attempt %d', testcase_name, i)
+
+                    test_runtime_secs = 10
+                    time.sleep(test_runtime_secs)
+                    after_stats = get_client_accumulated_stats()
+
+                    success = True
+                    for status, pct in expected_results.items():
+                        rpc = 'UNARY_CALL'
+                        qty = (
+                            after_stats.stats_per_method[rpc].result[status] -
+                            before_stats.stats_per_method[rpc].result[status])
+                        want = pct * args.qps * test_runtime_secs
+                        # Allow 10% deviation from expectation to reduce flakiness
+                        VARIANCE_ALLOWED = 0.1
+                        if abs(qty - want) > want * VARIANCE_ALLOWED:
+                            logger.info(
+                                '%s: failed due to %s[%s]: got %d want ~%d',
+                                testcase_name, rpc, status, qty, want)
+                            success = False
+                    if success:
+                        logger.info('success')
+                        break
```

Comment:
Seems this break only breaks from the inner loop, but not the outer `While True`.